File: src/main/java/main/utils/locale/LocaleConfig.java
package main.utils.locale;

import main.utils.json.AbstractGuildConfig;
import net.dv8tion.jda.api.entities.Guild;

public class LocaleConfig extends AbstractGuildConfig {
    private final Guild guild;
    private final long gid;

    public LocaleConfig(Guild guild) {
        super(guild);
        this.guild = guild;
        this.gid = guild.getIdLong();
    }

    public void setLocale(RobertifyLocale locale) {
        final var guildObject = getGuildObject();
        guildObject.put(Field.LOCALE.toString().toLowerCase(), locale.name().toLowerCase());
        getCache().updateGuild(guildObject);
    }

    public RobertifyLocale getLocale() {
        final var guildObject = getGuildObject();
        if (guildObject.has(Field.LOCALE.toString().toLowerCase()))
            return RobertifyLocale.parse(guildObject.getString(Field.LOCALE.toString().toLowerCase()));
        return null;
    }

    @Override
    public void update() {

    }

    public enum Field {
        LOCALE
    }
}

File: src/main/kotlin/main/utils/json/locale/LocaleConfig.kt
package main.utils.json.locale

import main.utils.json.AbstractGuildConfig
import main.utils.locale.RobertifyLocale
import net.dv8tion.jda.api.entities.Guild

class LocaleConfig(private val guild: Guild) : AbstractGuildConfig(guild) {

    fun getLocale(): RobertifyLocale {
        return RobertifyLocale.parse(getGuildModel().locale)
    }

    fun setLocale(locale: RobertifyLocale) {
        cache.updateGuild(guild.id) {
            this.locale = locale.name.lowercase()
        }
    }

    override fun update() {
        TODO("Not yet implemented")
    }


    enum class Field {
        LOCALE;

        override fun toString(): String = name.lowercase()
    }

}